components/admin_console/manage_teams_modal/remove_from_team_button.jsx

```diff
@@ -5,32 +5,30 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
-import {removeUserFromTeam} from 'actions/team_actions.jsx';
-
 export default class RemoveFromTeamButton extends React.PureComponent {
     static propTypes = {
         onError: PropTypes.func.isRequired,
         onMemberRemove: PropTypes.func.isRequired,
         team: PropTypes.object.isRequired,
         user: PropTypes.object.isRequired,
+        removeUserFromTeam: PropTypes.func.isRequired,
     };
 
     constructor(props) {
         super(props);
 
-        this.handleClick = this.handleClick.bind(this);
         this.handleMemberRemove = this.handleMemberRemove.bind(this);
     }
 
-    handleClick(e) {
+    handleClick = async (e) => {
         e.preventDefault();
 
-        removeUserFromTeam(
-            this.props.team.id,
-            this.props.user.id,
-            this.handleMemberRemove,
-            this.props.onError
-        );
+        const {data, error} = await this.props.removeUserFromTeam(this.props.team.id, this.props.user.id);
```

Comment:
this function and the one in `manage_teams_dropdown.jsx` basically does the same thing, perhaps you can combine it into one function and pass that as just one prop?

Reply:
I don't follow what you meant as passing them as one prop here. Could you help out with some pseudo logic here? 

`handleMemberRemove` is a prop and it has some state associated in the parent so we need a callback on success. same case with onError

Reply:
0/5 here but the parent component could pass a prop like `handleRemoveUserFromTeam` then in the parent component you have

```js
async handleRemoveUserFromTeam() {
    const {actions, user, team} = this.props;

    const {data, error} = actions.removeUserFromTeam(team.id, user.id);
     if (data) {
       this.handleMemberRemove();
    } else if (error) {
       this.handleError(error.message);
    }
}
```

then for both components you should only need a prop for `handleRemoveUserFromTeam` and you can remove the props for `removeUserFromTeam`, `onMemberRemove` and `onError`

Reply:
Nice call. Made changes accordingly.